clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -354,6 +358,494 @@ fatbinary(ArrayRef<std::pair<StringRef, StringRef>> InputFiles,
 }
 } // namespace amdgcn
 
+namespace sycl {
```

Comment:
Hi @AlexeySachkov 
There is considerable interaction between the 'sycl' portion and the remaining portion. For example, I do rely on the logic to create temporary files, several static variables defined in this file, and error reporting logic. It might require considerable rework to achieve this. Since I am encapsulating all these changes inside a namespace, I think it should be reasonable to manage this.

Thanks

Reply:
Ok, works for me for the initial version. I imagine that amount of such dependencies could be reduced going forward when we switch to library calls instead of external processes for most of the functionality.